File: soln-string-problems/palindrome-longest-build.java
/**
 @author Farheen Bano
  
 Reference-
 https://leetcode.com/problems/longest-palindrome/
*/

import java.util.*;
import java.lang.*;
import java.io.*;

class Solution {
    public int longestPalindrome(String s) {
        HashMap<Character,Integer> map=new HashMap<>();
        char[] char_s=s.toCharArray();
        
        for(char ch:char_s){
            map.put(ch,map.getOrDefault(ch,0)+1);
        }
        
        int flag=0;
        for(char key:map.keySet()){
            if(map.get(key)%2!=0){
                flag++;
            }
        }
        return flag>0?s.length()-flag+1:s.length();
    }
}

File: C++/soln-string-problems/palindrome-longest-build.cpp
/**
 @author Farheen Bano
  
 Date 16-07-2021

 Reference-
 https://leetcode.com/problems/longest-palindrome/
*/

class Solution {
public:
    int longestPalindrome(string s) {
        unordered_map<char,int> map;
        
        for(char ch:s)
            map[ch]++;
        
        int flag=0;
        for(auto element:map){
            if(element.second%2!=0){
                flag++;
            }
        }
        return flag>0?s.size()-flag+1:s.size();
    }
};